doc/BACKUP.md

```diff
@@ -0,0 +1,400 @@
+
+More sophisticated setups with more than two devices are possible.
+Take note that "RAID 1" in `btrfs` means "data is copied on up to two
+devices", meaning only up to one device can fail.
+You may be interested in `raid1c3` and `raid1c4` modes if you have
+three or four storage devices.
+BTRFS would probably work better if you were purchasing an entire set
+of new storage devices to set up a new node.
+
+`backup` Plugin And Remote NFS Mount
+------------------------------------
+
+You can get the `backup` plugin here:
+https://github.com/lightningd/plugins/tree/master/backup
+
+The `backup` plugin requires Python 3.
+
+* `cd` into its directory and install requirements.
+  * `pip3 install -r requirements.txt`
+* Figure out where you will put the backup files.
+  * Ideally you have an NFS or other network-based mount on your system,
+    into which you will put the backup.
+* Stop your Lightning node.
+* `/path/to/backup-cli init ${LIGHTNINGDIR} file:///path/to/nfs/mount`.
+  This creates an initial copy of the database at the NFS mount.
+* Add these settings to your `lightningd` configuration:
+  * `important-plugin=/path/to/backup.py`
+  * `backup-destination=file:///path/to/nfs/mount`
+* Restart your Lightning node.
+
+It is recommended that you use a network-mounted filesystem for the backup
+destination.
+
+Alternately, you *could* put it in another storage device (e.g. USB flash
+disk) in the same physical location, but if so, you might want to use the
+above `btrfs` RAID-1 instead, as that tends to be more performant
+(though using this `backup` plugin does not require you to repartition an
+existing hard disk, which might matter to you if you are using an
+everyday-use laptop or desktop).
+
+To recover:
+
+* Re-download the `backup` plugin and install Python 3 and the
+  requirements of `backup`.
+* `/path/to/backup-cli restore file:///path/to/nfs/mount ${LIGHTNINGDIR}`
+
+If your backup destination is a network-mounted filesystem that is in a
+remote location, then even loss of all hardware in one location will allow
+you to still recover your Lightning funds.
+
+However, if instead you are just replicating the database on another
+storage device in a single location, you remain vulnerable to disasters
+like fire or computer confiscation.
+
+PostgreSQL Cluster
+------------------
+
+`lightningd` may also be compiled with PostgreSQL support.
+PostgreSQL is generally faster than SQLITE3, and also supports running a
+PostgreSQL cluster to be used by `lightningd`, with automatic replication
+and turnover in case an entire node of the PostgreSQL cluster fails.
+
+Setting this up, however, is more involved.
+
+By default, `lightningd` compiles without PostgreSQL support.
+To enable it, you have to install a developer version of `libpq`.
+On most Debian-derived systems that would be `libpq-dev`.
+To verify you have it properly installed on your system, check if the
+following command gives you a path:
+
+    pg_config --includedir
+
```

Comment:
I didn't know it. When I built it, everything worked out of the box. Pretty useful to know. 
I probably had already the library installed.

Reply:
Okay, maybe the Lightning+PostgreSQL can be a separate PR?

Reply:
The document you mean?
Something like `doc/postgresQL.md`?

Anyway This part in the backup strategies seems pretty right to me. At least to mention a more extended part in the doc directory.  @fiatjaf Has thought about that?

Reply:
Yes, the document I mean.